### PR TITLE
feat(db_common): Add Flutter library

### PR DESCRIPTION
**Pull request by dnys1**
Adds a Flutter library to replace `sqlite3_flutter_libs` in the short-term until we remove our dependency on iOS DataStore.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -133,30 +133,69 @@ jobs:
...
-        working-directory: packages/auth/amplify_auth_cognito
+      - name: "packages/analytics/amplify_analytics_pinpoint/example/ios/.symlinks/plugins/amplify_secure_storage; dart format --output=none --set-exit-if-changed ."
+        if: "always() && steps.packages_analytics_amplify_analytics_pinpoint_example_ios_.symlinks_plugins_amplify_secure_storage_pub_upgrade.conclusion == 'success'"
+        working-directory: packages/analytics/amplify_analytics_pinpoint/example/ios/.symlinks/plugins/amplify_secure_storage
         run: "dart format --output=none --set-exit-if-changed ."
-      - name: "packages/auth/amplify_auth_cognito; flutter analyze --fatal-infos lib"
-        if: "always() && steps.packages_auth_amplify_auth_cognito_pub_upgrade.conclusion == 'success'"
-        working-directory: packages/auth/amplify_auth_cognito
-        run: flutter analyze --fatal-infos lib
+      - name: "packages/analytics/amplify_analytics_pinpoint/example/ios/.symlinks/plugins/amplify_secure_storage; flutter analyze --fatal-infos lib pigeons"
```
Comment (fjnoyp) [inline .github/workflows/lint.yml:145]: Seems weird we're adding all these changes for just adding flutter library?  This seems to be unrelated to the PR, perhaps we could add a comment explaining what additional changes were made here as well.

File: packages/common/amplify_db_common/LICENSE [hunk 2]
```diff
@@ -1,175 +1 @@
-
```
Comment (fjnoyp) [inline packages/common/amplify_db_common/LICENSE:1]: Shouldn't this LICENSE be the same as well?

File: packages/common/amplify_db_common/android/.gitignore [hunk 3]
```diff
@@ -0,0 +1,9 @@
+*.iml
+.gradle
```
Comment (fjnoyp) [inline packages/common/amplify_db_common/android/.gitignore:2]: Why are we adding android, ios, linux, macos, windows folders outside of the example sub directory?

File: packages/common/amplify_db_common/example/lib/db.dart [hunk 4]
```diff
@@ -0,0 +1,61 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (fjnoyp) [inline packages/common/amplify_db_common/example/lib/db.dart:1]: Not a major issue, but is there a way to change your commit so that this is a rename?  These files were created by Jordan but are just showing up as new.

File: packages/common/amplify_db_common/example/test/widget_test.dart [hunk 5]
```diff
@@ -0,0 +1,41 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
```
Comment (fjnoyp) [inline packages/common/amplify_db_common/example/test/widget_test.dart:2]: Is this file necessary?

File: packages/common/amplify_db_common/pubspec.yaml [hunk 6]
```diff
@@ -1,29 +1,35 @@
...
+flutter:
+  plugin:
+    platforms:
+      android:
+        package: com.amazonaws.amplify.amplify_db_common
+        pluginClass: AmplifyDbCommonPlugin
+      linux:
+        pluginClass: AmplifyDbCommonPlugin
+      windows:
+        pluginClass: AmplifyDbCommonPlugin
```
Comment (fjnoyp) [inline packages/common/amplify_db_common/pubspec.yaml:35]: what about MacOS?

File: packages/common/amplify_db_common/README.md [hunk 7]
```diff
@@ -1,15 +1,15 @@
-## amplify_db_common
+# amplify_db_common
```
Comment (fjnoyp) [inline packages/common/amplify_db_common/README.md:2]: Can we at least mention this is temporary here to keep track.  In case the team forgets this is a temporary fix.

Comment (fjnoyp) [reply packages/common/amplify_db_common/README.md:2]: This top level Flutter library will replace our import of sqlite3_flutter_libs in the short term until we remove our dependency on iOS DataStore.  When that happens we will switch to using sqlite3_flutter_libs and the amplify_db_common_dart package instead.

File: packages/common/amplify_db_common/android/.gitignore (hunk 3 above)
Comment (fjnoyp) [reply packages/common/amplify_db_common/android/.gitignore:2]: I think they were autogenerated but have no use in our context here right?  If so can we remove it to reduce the # of files being added in the PR.

File: packages/common/amplify_db_common/ios/amplify_db_common.podspec [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+  s.name             = 'amplify_db_common'
+  s.version          = '0.0.1'
+  s.summary          = 'A new Flutter plugin project.'
+  s.description      = <<-DESC
+A new Flutter plugin project.
+                       DESC
+  s.homepage         = 'http://example.com'
+  s.license          = { :file => '../LICENSE' }
+  s.author           = { 'Your Company' => 'email@example.com' }
+
```
Comment (HuiSF) [inline packages/common/amplify_db_common/ios/amplify_db_common.podspec:15]: Should these fields be updated?

File: packages/common/amplify_db_common/macos/amplify_db_common.podspec [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+Pod::Spec.new do |s|
+  s.name             = 'amplify_db_common'
+  s.version          = '0.0.1'
+  s.summary          = 'A new Flutter plugin project.'
+  s.description      = <<-DESC
+A new Flutter plugin project.
+                       DESC
+  s.homepage         = 'http://example.com'
+  s.license          = { :file => '../LICENSE' }
+  s.author           = { 'Your Company' => 'email@example.com' }
```
Comment (HuiSF) [inline packages/common/amplify_db_common/macos/amplify_db_common.podspec:14]: ditto

**Review by HuiSF — APPROVED** [review]

File: packages/common/amplify_db_common/lib/src/connect_io.dart [hunk 10]
```diff
@@ -12,55 +12,21 @@
...
-/// See [Using drift in a background isolate](https://drift.simonbinder.eu/docs/advanced-features/isolates/#initialization-on-the-main-thread)
-/// and [the drift native example](https://github.com/simolus3/drift/blob/3253cd7ead51d7b163542929eb0fc8daee573fce/examples/app/lib/database/connection/native.dart)
-/// for more info.
 QueryExecutor connect({
   required String name,
   String? path,
 }) {
-  assert(path != null, 'path cannot be null on vm.');
-
-  return DatabaseConnection.delayed(
```
Comment (fjnoyp) [inline packages/common/amplify_db_common/lib/src/connect_io.dart:35]: I was able to get drift working on iOS with this same code.  Is it necessary that we have the amplify_db_common and amplify_db_common_dart?  

Couldn't we just have amplify_sqlite3_flutter_libs (which would be sqlite3_flutter_libs without iOS) and then the existing amplify_db_common?

**Review by fjnoyp — COMMENTED** [review]
Thanks again for tackling this.  Added some comments.

File: packages/common/amplify_db_common/ios/amplify_db_common.podspec (hunk 8 above)
Comment (fjnoyp) [reply packages/common/amplify_db_common/ios/amplify_db_common.podspec:15]: I think the git change was weird, the file was actually moved but git registered it as modified.

File: packages/common/amplify_db_common/pubspec.yaml (hunk 6 above)
Comment (dnys1) [reply packages/common/amplify_db_common/pubspec.yaml:35]: Not needed for iOS/macOS - we just want the podspec. Probably not the others either, but I'm not sure how to remove them yet.

File: packages/common/amplify_db_common/README.md (hunk 7 above)
Comment (dnys1) [reply packages/common/amplify_db_common/README.md:2]: I'd rather not publish something saying it's a temporary fix. However, this package will be needed even once we decide to use the flutter_libs package. It incorporates path_provider into the `connect` function.

File: packages/common/amplify_db_common/lib/src/connect_io.dart (hunk 10 above)
Comment (dnys1) [reply packages/common/amplify_db_common/lib/src/connect_io.dart:35]: This adds path_provider to the connect function. The separation is nice if we ever need it for other things as well. A bit of overhead but I don't think it's too bad.

We don't want to bring in DataStore as a dependency though? We just want to match the SQLite version used which does require a separate package.

File: packages/common/amplify_db_common/macos/amplify_db_common.podspec (hunk 9 above)
Comment (dnys1) [reply packages/common/amplify_db_common/macos/amplify_db_common.podspec]: Updated

File: packages/common/amplify_db_common/ios/amplify_db_common.podspec (hunk 8 above)
Comment (dnys1) [reply packages/common/amplify_db_common/ios/amplify_db_common.podspec:16]: Updated.

File: packages/common/amplify_db_common/example/test/widget_test.dart (hunk 5 above)
Comment (dnys1) [reply packages/common/amplify_db_common/example/test/widget_test.dart]: Nope. Removed.

File: .github/workflows/lint.yml (hunk 1 above)
Comment (dnys1) [reply .github/workflows/lint.yml]: Didn't mean to - it's mono_repo being silly

**Review by fjnoyp — APPROVED** [review]
Looks good, approved.
